### ISSUE TITLE
Please have an easier toterial for installing on Linux Mint

### ISSUE THREAD

**Issue by 2000sLegacyOS** [rozniak/xfce-winxp-tc#233]
I can't figure out how to install this since the last guide on YouTube is out of date could you please go throw with me to install Linux Mint (Xfce) please like cause I've been trying to figure this out for almost two weeks now.

**Comment by rozniak** [comment]
Yeah that video is quite old by now... it was back before I really had much sorted out regarding building the project.

Nowadays it's pretty simple - have you read the steps here? https://github.com/rozniak/xfce-winxp-tc/tree/master/packaging#tldr-on-building

**Comment by 2000sLegacyOS** [comment]
It didnt work for me I'm on Linux Mint Xfce but it still didn't chnage the theme and I don't see it any where when I tired to change the theme.

**Comment by rozniak** [comment]
Did you install the packages that were outputted?

**Comment by 2000sLegacyOS** [comment]
I don't know I'm on an Ubuntu version of Linux mint which Ubuntu is still Debian for all I know but all I did was followed the TL;DR it didn't go as planned.

**Comment by rozniak** [comment]
When you run `buildall.sh` mate it compiles and packages everything, and puts them in the `xptc` folder in the same directory. If you look in there you should see a bunch of `.deb` files -- those are what you are gonna want to install.

They *do* have an order to installing them, but you can 'cheat' and just keep trying to install the whole lot until you get no errors by doing:

`find . -type f -iname "*.deb" -exec sudo dpkg -i '{}' \;`

You can keep running that in the `xptc` dir until it looks like everything installed okay.

**Comment by 2000sLegacyOS** [comment]
This is what I got when I ran it I also have screenshots:
Selecting previously unselected package wintc-sound-theme-xp.
(Reading database ... 529464 files and directories currently installed.)
Preparing to unpack ./wintc-sound-theme-xp.deb ...
Unpacking wintc-sound-theme-xp (1.0) ...
Setting up wintc-sound-theme-xp (1.0) ...
Selecting previously unselected package libwintc-shllang.
(Reading database ... 529521 files and directories currently installed.)
Preparing to unpack ./libwintc-shllang.deb ...
Unpacking libwintc-shllang (1.0) ...
dpkg: dependency problems prevent configuration of libwintc-shllang:
 libwintc-shllang depends on libwintc-comgtk; however:
  Package libwintc-comgtk is not installed.

dpkg: error processing package libwintc-shllang (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libwintc-shllang
Selecting previously unselected package wintc-notepad.
(Reading database ... 529602 files and directories currently installed.)
Preparing to unpack ./wintc-notepad.deb ...
Unpacking wintc-notepad (1.0) ...
dpkg: dependency problems prevent configuration of wintc-notepad:
 wintc-notepad depends on libwintc-comgtk; however:
  Package libwintc-comgtk is not installed.
 wintc-notepad depends on libwintc-shllang; however:
  Package libwintc-shllang is not configured yet.

dpkg: error processing package wintc-notepad (--install):
 dependency problems - leaving unconfigured
Processing triggers for gnome-menus (3.36.0-1ubuntu3) ...
Processing triggers for desktop-file-utils (0.26+mint3+victoria) ...
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Errors were encountered while processing:
 wintc-notepad
Selecting previously unselected package libwintc-shelldpa.
(Reading database ... 529639 files and directories currently installed.)
Preparing to unpack ./libwintc-shelldpa.deb ...
Unpacking libwintc-shelldpa (1.0) ...
dpkg: dependency problems prevent configuration of libwintc-shelldpa:
 libwintc-shelldpa depends on libwintc-comgtk; however:
  Package libwintc-comgtk is not installed.

dpkg: error processing package libwintc-shelldpa (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libwintc-shelldpa
Selecting previously unselected package wintc-theme-native.
(Reading database ... 529644 files and directories currently installed.)
Preparing to unpack ./wintc-theme-native.deb ...
Unpacking wintc-theme-native (1.0) ...
Setting up wintc-theme-native (1.0) ...
Selecting previously unselected package wintc-cursor-theme-standard-no-shadow.
(Reading database ... 529914 files and directories currently installed.)
Preparing to unpack .../wintc-cursor-theme-standard-no-shadow.deb ...
Unpacking wintc-cursor-theme-standard-no-shadow (1.0) ...
Setting up wintc-cursor-theme-standard-no-shadow (1.0) ...
Selecting previously unselected package wintc-cursor-theme-standard-with-shadow.
(Reading database ... 530059 files and directories currently installed.)
Preparing to unpack .../wintc-cursor-theme-standard-with-shadow.deb ...
Unpacking wintc-cursor-theme-standard-with-shadow (1.0) ...
Setting up wintc-cursor-theme-standard-with-shadow (1.0) ...
Selecting previously unselected package libwintc-comgtk.
(Reading database ... 530204 files and directories currently installed.)
Preparing to unpack ./libwintc-comgtk.deb ...
Unpacking libwintc-comgtk (1.0) ...
Setting up libwintc-comgtk (1.0) ...
Selecting previously unselected package wintc-theme-professional.
(Reading database ... 530209 files and directories currently installed.)
Preparing to unpack ./wintc-theme-professional.deb ...
Unpacking wintc-theme-professional (1.0) ...
Setting up wintc-theme-professional (1.0) ...
Selecting previously unselected package libwintc-winbrand.
(Reading database ... 530508 files and directories currently installed.)
Preparing to unpack ./libwintc-winbrand.deb ...
Unpacking libwintc-winbrand (1.0) ...
Setting up libwintc-winbrand (1.0) ...
Selecting previously unselected package wintc-bootvid.
(Reading database ... 530517 files and directories currently installed.)
Preparing to unpack ./wintc-bootvid.deb ...
Unpacking wintc-bootvid (1.0) ...
Setting up wintc-bootvid (1.0) ...
Selecting previously unselected package wintc-taskband.
(Reading database ... 530523 files and directories currently installed.)
Preparing to unpack ./wintc-taskband.deb ...
Unpacking wintc-taskband (1.0) ...
dpkg: dependency problems prevent configuration of wintc-taskband:
 wintc-taskband depends on libwintc-exec; however:
  Package libwintc-exec is not installed.
 wintc-taskband depends on libwintc-shelldpa; however:
  Package libwintc-shelldpa is not configured yet.
 wintc-taskband depends on libwintc-shllang; however:
  Package libwintc-shllang is not configured yet.

dpkg: error processing package wintc-taskband (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 wintc-taskband
Selecting previously unselected package wintc-shell-run.
(Reading database ... 530560 files and directories currently installed.)
Preparing to unpack ./wintc-shell-run.deb ...
Unpacking wintc-shell-run (1.0) ...
dpkg: dependency problems prevent configuration of wintc-shell-run:
 wintc-shell-run depends on libwintc-comctl; however:
  Package libwintc-comctl is not installed.
 wintc-shell-run depends on libwintc-exec; however:
  Package libwintc-exec is not installed.
 wintc-shell-run depends on libwintc-shllang; however:
  Package libwintc-shllang is not configured yet.

dpkg: error processing package wintc-shell-run (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 wintc-shell-run
Selecting previously unselected package libwintc-exec.
(Reading database ... 530593 files and directories currently installed.)
Preparing to unpack ./libwintc-exec.deb ...
Unpacking libwintc-exec (1.0) ...
Setting up libwintc-exec (1.0) ...
Selecting previously unselected package wintc-icon-theme-luna.
(Reading database ... 530598 files and directories currently installed.)
Preparing to unpack ./wintc-icon-theme-luna.deb ...
Unpacking wintc-icon-theme-luna (1.0) ...
Setting up wintc-icon-theme-luna (1.0) ...
gtk-update-icon-cache: Cache file created successfully.
Selecting previously unselected package wintc-shell-winver.
(Reading database ... 532434 files and directories currently installed.)
Preparing to unpack ./wintc-shell-winver.deb ...
Unpacking wintc-shell-winver (1.0) ...
dpkg: dependency problems prevent configuration of wintc-shell-winver:
 wintc-shell-winver depends on libwintc-comctl; however:
  Package libwintc-comctl is not installed.

dpkg: error processing package wintc-shell-winver (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 wintc-shell-winver
Selecting previously unselected package wintc-theme-luna-blue.
(Reading database ... 532436 files and directories currently installed.)
Preparing to unpack ./wintc-theme-luna-blue.deb ...
Unpacking wintc-theme-luna-blue (1.0) ...
Setting up wintc-theme-luna-blue (1.0) ...
Selecting previously unselected package wintc-fonts-xp.
(Reading database ... 532802 files and directories currently installed.)
Preparing to unpack ./wintc-fonts-xp.deb ...
Unpacking wintc-fonts-xp (1.0) ...
Setting up wintc-fonts-xp (1.0) ...
Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...
Selecting previously unselected package wintc-desktop.
(Reading database ... 532858 files and directories currently installed.)
Preparing to unpack ./wintc-desktop.deb ...
Unpacking wintc-desktop (1.0) ...
dpkg: dependency problems prevent configuration of wintc-desktop:
 wintc-desktop depends on libwintc-shelldpa; however:
  Package libwintc-shelldpa is not configured yet.

dpkg: error processing package wintc-desktop (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 wintc-desktop
Selecting previously unselected package wintc-build-tag.
(Reading database ... 532859 files and directories currently installed.)
Preparing to unpack ./wintc-build-tag.deb ...
Unpacking wintc-build-tag (1.0) ...
Setting up wintc-build-tag (1.0) ...
Selecting previously unselected package libwintc-comctl.
(Reading database ... 532860 files and directories currently installed.)
Preparing to unpack ./libwintc-comctl.deb ...
Unpacking libwintc-comctl (1.0) ...
Setting up libwintc-comctl (1.0) ...

![XP Term1](https://github.com/rozniak/xfce-winxp-tc/assets/150536992/59a907b3-b5dc-41ac-8c3b-2643e474b940)
![XP Term 2](https://github.com/rozniak/xfce-winxp-tc/assets/150536992/635dfcb0-db3f-4d0b-9e18-6cf6dd1fbaec)
![XP Term 3](https://github.com/rozniak/xfce-winxp-tc/assets/150536992/25e45d2c-f5a0-44d8-af2d-862834c7ee5e)
![XP Term 4](https://github.com/rozniak/xfce-winxp-tc/assets/150536992/9d14320c-be27-4cfd-a9cd-3c46c8503e86)
![XP messed up](https://github.com/rozniak/xfce-winxp-tc/assets/150536992/65966486-b9fe-47ce-a60a-dc6b249d7eb3)

It doesn't look right what should I do?

**Comment by 2000sLegacyOS** [comment]
I tried that multiple times I have  grassmunk's Chicago95 installed and when I try to switch it to any other theme it doesn't work so do you think they could be clashing with something in my system? https://github.com/grassmunk/Chicago95/releases/tag/v3.0.1

**Comment by 2000sLegacyOS** [comment]
I might have to reinstall my system.

**Comment by rozniak** [comment]
I don't think anything should be clashing with this repo, everything installs as system-wide packages. Most things are prefixed with `wintc-`, and the things that aren't should have unique names (unless you have some other desktop theme called `Windows XP Style (Blue)`?

If you ran that command multiple times, the later runs should be more successful. Essentially the errors are dependency issues - which should be 'resolved' if you keep trying to install the whole lot, each time it will have installed more successfully. (this is a bodge but it works for now)

What issues in particular are you having when switching themes? Certain programs aren't themed? Are they glitchy? :thinking:

**Comment by 2000sLegacyOS** [comment]
No I even tried to switch back to Linux Mint't Default theme and my system seems kinda borked from Chicago95?

Also can you please put this in the Synaptic Package Manager as a Package for Debian and Ubuntu based Distros which includes Linux Mint please I think it would make installs alot less buggier I was up the whole night trying to install this but when I tried to change back from Chicago95 theme everything in the system was not changing for some reason so I might have to do a clean install.

**Comment by rozniak** [comment]
> No I even tried to switch back to Linux Mint't Default theme and my system seems kinda borked from Chicago95?

I have no idea man, I have never used Chicago95.

> Also can you please put this in the Synaptic Package Manager as a Package for Debian and Ubuntu based Distros which includes Linux Mint please

I can't just put this project in upstream repos. It would not make a difference anyway, since Synaptic is just a front-end, the packages would be identical to the ones you built with the command.

**Comment by onkoe** [comment]
@2000sLegacyOS It seems likely that your appearance configurations are a bit messed up. I’d recommend the following:

1. Install GNOME Tweaks (`sudo apt install gnome-tweaks`
1. Open it, go to the Appearance tab, and select default options - Adwaita for Cursor, Icons, and Legacy Applications. Use “Default” for Sound.
1. Head to the Font tab and select Cantarell Regular 11 for the interface + document text, Source Code Pro Regular 10 for monospace, and Cantarell Bold 11 for legacy window titles.
1. Open Xfce Appearance Settings. Under the Style tab, hit Adwaita. Go to Icons and hit Adwaita. Under font, use Cantarell.
1. Try applying Roz’s `winxp-tc` theme again - it should look normal. Make sure to apply the panel patch if you’re looking for a ‘correct’ notification area. ☺️

**Comment by rozniak** [comment]
> Make sure to apply the panel patch if you’re looking for a ‘correct’ notification area.

FYI that panel patch is obsolete since `wintc-taskband` replaces XFCE panel nowadays. :stuck_out_tongue:

**Comment by 2000sLegacyOS** [comment]
Really someone needs to do a video on their YouTube channel about this @rozniak do you have a YouTube Channel I don't do go with these kind of stuff unless if its a Video so it makes it much easier for me to understand and follow what's going on.

**Comment by rozniak** [comment]
I haven't put together any video because I work on the project so much by the time the video is done it'll be obsolete within like 2 merges.

All I can recommend is perhaps trying things out in a new VM to at least get things working before faffing around with mixing and matching.

**Comment by 2000sLegacyOS** [comment]
Okay I will do that.

**Comment by 2000sLegacyOS** [comment]
Okay I think though it also being a in Pack manger or software center would help alot even if its upstream it could make it alot easier to update when it comes to that as well.

**Comment by onkoe** [comment]
Unfortunately, this set of customizations is unlikely to ever be packaged for upstream usage. 🥹

Most Linux distributions tend to keep packages limited to a specific scope, particularly watching for packages that might be buggy or legally problematic.

This project could flag both of those. It touches on stability with it being a customization pack that changes a lot of values across the system. It modifies the panel and optionally replaces it altogether with the taskband, altering system files in the process. Xfce also faces a lot of difficult-to-reverse configuration changes.

On the other hand, this project uses some assets from Windows XP. Being a passion project with community support, it’s unlikely that anyone would be upset about it as-is. However, if packaged upstream, Debian/Ubuntu/Mint would all have to deal with copyrighted assets being redistributed in their repos. Without a lot of asset creation, this project would be toast! 🍞

I hope this helps! ☺️

**Comment by 2000sLegacyOS** [comment]
Yeah it does I wish it was much easier to install without a YouTube Video on guiding you though it, also I really wish there was an really user friendly and noob friendly way to install this without much hassle like a single command in the CLI or just double-clicking .deb package for those of us on Debian/Ubuntu/Mint or anything related to them to make it way to just get this on the device without much trouble.

But I get it I just hope with-in a couple of merge request or two that this problem gets fixed all-together or just mostly so a user can just install it and not have to worry about it.

**Comment by 2000sLegacyOS** [comment]
I wonder if the next release will be easier to install with less problems and much more user-friendly?

**Comment by rozniak** [comment]
'Next release' implies there has been a formal release at all so far, which there hasn't. :stuck_out_tongue: 

#81 exists for doing setup/installer work, but I only have one pair of hands - it isn't really a priority at the moment because things aren't complete enough to warrant an installer imo. If I wrote one I'd probably get a bunch of issues opened about how there's no icons in the tray, no taskmgr, no explorer, Start menu doesn't show most frequently used etc.

**Comment by 2000sLegacyOS** [comment]
Okay then I guess I will have to do it again.

**Comment by rozniak** [comment]
Let us know how it goes mate! Any troubles just reply here I'm happy to help. :sunglasses:
